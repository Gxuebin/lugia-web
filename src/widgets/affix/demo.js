/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Affix from './affix';
import Button from '../button';

export default class AffixDemo extends React.Component {
  constructor() {
    super();
    this.EleRef = null;
    this.Ele2Ref = null;
    this.state = {
      display: true,
    };
  }
  render() {
    const { display } = this.state;
    return (
      <div>
        {display ? (
          <div style={{ width: '200px', height: '1200px' }}>
            <div style={{ width: '200px', height: '200px' }} />
            <Affix offsetTop={50}>
              <Button>affix-top</Button>
            </Affix>

            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.EleRef = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '100px' }} />
                <Affix offsetTop={50} target={() => this.EleRef}>
                  <Button>affix-top</Button>
                </Affix>
              </div>
            </div>
            <div
              style={{ width: '200px', height: '200px', overflowY: 'scroll' }}
              ref={node => (this.Ele2Ref = node)}
            >
              <div style={{ height: '400px' }}>
                <div style={{ width: '20px', height: '200px' }} />
                <Affix offsetBottom={50} target={() => this.Ele2Ref}>
                  <Button>affix2-bottom</Button>
                </Affix>
              </div>
            </div>

            <div style={{ width: '200px', height: '500px' }} />
            <Affix offsetBottom={50}>
              <Button>affix-bottom</Button>
            </Affix>
            <div style={{ width: '200px', height: '500px' }} />
          </div>
        ) : (
          <div style={{ width: '200px', height: '1200px' }} />
        )}
      </div>
    );
  }
}
