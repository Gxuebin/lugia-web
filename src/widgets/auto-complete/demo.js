import * as React from 'react';
import AutoComplete from './';

const data = [
  'Nikcy Romero',
  'Armin van Buuren',
  'Hardwell',
  'Zedd',
  'Kazze',

  'Vicetone',
  'Martin Garrix',
  'David Guetta',
  'The Chainsmokers',
  'Kygo',

  'Axwell ^ Ingrosso',
  'Dimitri Vegas & Like Mike',
  'Calvin Harris',
  'Avicci',
  'Fedde Le Grand',

  'Tiesto',
  'Snoop Dogg',
  'Bassjackers',
  'Sebastian Ingrosso',
  'Swedish House Mafia',
  'Alesso',
  'Afrojack',
  'Knife Party',
  'Dannic',
  'R3hab',
];

export default class AutoCompleteBounded extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      menuData: data,
    };
  }
  static getDerivedStateFromProps(props: any, state: any) {
    const hasValueInProps = 'value' in props;
    const value = hasValueInProps ? props.value : state ? state.value : props.defaultValue;
    if (!state) {
      return {
        value,
      };
    }
    return {
      value,
    };
  }

  render() {
    const { menuData, value } = this.state;
    return <AutoComplete showOldValue value={value} data={menuData} onChange={this.onChange} />;
  }

  onChange = (value: string) => {
    this.search(value);
    this.setState({ value });
  };

  search(query: string) {
    let menuData;
    let rowSet = [];
    const len = data.length;

    for (let i = 0; i < len; i++) {
      const row = data[i];
      if (this.searchValue(query, row)) {
        rowSet.push(row);
      }
      if (query === row) {
        rowSet = [];
        break;
      }
    }

    if (rowSet.length === len) {
      menuData = data;
    } else {
      menuData = rowSet;
    }
    this.setState({ menuData });
  }

  searchValue = (query: string, row: string): boolean => {
    return row.indexOf(query) !== -1;
  };
}

// export default class Demo extends React.Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       menuData: data,
//       value: '',
//     };
//   }
//   render() {
//     return (
//       <AutoComplete
//         data={this.state.menuData}
//         onChange={this.onChange}
//         placeholder={'请输入'}
//         showOldValue={true}
//       />
//     );
//   }

//   onChange = value => {
//     this.search(value);
//   };

//   search(query) {
//     let menuData;
//     let rowSet = [];
//     const len = data.length;

//     for (let i = 0; i < len; i++) {
//       const row = data[i];
//       if (query === row) {
//         rowSet = [];
//         break;
//       }

//       if (this.searchValue(query, row)) {
//         rowSet.push(row);
//       }
//     }

//     if (rowSet.length === len) {
//       menuData = data;
//     } else {
//       menuData = rowSet;
//     }
//     this.setState({ menuData });
//   }

//   searchValue = (query, row) => {
//     return row.indexOf(query) !== -1 || row === query;
//   };
// }
