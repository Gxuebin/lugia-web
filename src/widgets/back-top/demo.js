/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';

const Demo = styled.div`
  & > span > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        normal: {
          color: 'red',
          background: { backgroundColor: 'orange' },
          width: 100,
          height: 100,
          borderRadius: 100,
          opacity: 0.7,
          border: {
            borderWidth: 1,
            borderStyle: 'solid',
            borderColor: 'red',
          },
        },
      },
    };
    return (
      <div style={{ height: '1600px' }}>
        <BackTop />
        <Demo cur={1}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop />
          </Theme>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={3}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop visibilityHeight={80} target={() => this.EleRef} />
          </Demo>
        </div>
      </div>
    );
  }
}
