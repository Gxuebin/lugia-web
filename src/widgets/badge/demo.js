/**
 *
 * create by liangguodong on 2018/8/27
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import Badge from './';
import { Col, Row } from '../grid';
import Widget from '../consts';
import Theme from '../theme';
import Icon from '../icon';

const Box = styled.div`
  width: 40px;
  height: 40px;
  background: #ccc;
  margin-left: 10px;
`;
const Wrapper = styled.div`
  margin-left: 10px;
  margin-top: 50px;
`;

class Turn extends React.Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  click = type => () => {
    const count = type === 'plus' ? this.state.count + 1 : this.state.count - 1;
    this.setState({ count });
  };

  render() {
    const config = {
      [Widget.Badge]: {
        children: {
          [Widget.NumberTurn]: {
            normal: { position: { top: -5, right: -5 } },
          },
        },
      },
    };
    return (
      <Wrapper>
        <Theme config={config}>
          <Row>
            <Col span={4}>
              <Badge count={this.state.count} showZero overflowCount={99}>
                <Box />
              </Badge>
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_plus_square_o"
                onClick={this.click('plus')}
              />
              <Icon
                style={{ fontSize: '2em' }}
                iconClass="lugia-icon-reminder_minus_square_o"
                onClick={this.click('minus')}
              />
            </Col>
          </Row>
        </Theme>
      </Wrapper>
    );
  }
}

export default () => {
  const view = {
    [Widget.Badge]: {
      normal: {
        position: { top: -5, right: 0 },
        background: { backgroundColor: 'orange' },
        color: 'black',
      },
    },
  };
  return (
    <div>
      <Theme
        config={{
          [Widget.Badge]: {
            children: {
              [Widget.NumberTurn]: {
                normal: { position: { top: -5, right: -5 } },
              },
            },
          },
        }}
      >
        <Wrapper>
          <Row>
            <Col span={4}>
              <Badge count={4}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge count={99}>
                <Box />
              </Badge>
            </Col>
            <Col span={4}>
              <Badge>
                <Box />
              </Badge>
            </Col>
          </Row>
        </Wrapper>
        <Wrapper>
          <Theme config={view}>
            <Row>
              <Col span={4}>
                <Badge count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <Badge showZero count={0}>
                  <Box />
                </Badge>
              </Col>
              <Col span={4}>
                <a href="www.baidu.com">
                  <Badge>
                    <Box />
                  </Badge>
                </a>
              </Col>
            </Row>
          </Theme>
        </Wrapper>
        <Wrapper>
          <Row>
            <Col span={4}>
              <Theme
                config={{
                  green: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'green' },
                    },
                  },
                }}
              >
                <Badge viewClass="green">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  purple: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'purple' },
                    },
                  },
                }}
              >
                <Badge viewClass="purple">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  yellow: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'yellow' },
                    },
                  },
                }}
              >
                <Badge viewClass="yellow">
                  <Box />
                </Badge>
              </Theme>
            </Col>
            <Col span={4}>
              <Theme
                config={{
                  blue: {
                    normal: {
                      position: { top: -5, right: 2 },
                      background: { backgroundColor: 'blue' },
                    },
                  },
                }}
              >
                <Badge viewClass="blue">
                  <Box />
                </Badge>
              </Theme>
            </Col>
          </Row>
        </Wrapper>
        <Turn />
      </Theme>
    </div>
  );
};
