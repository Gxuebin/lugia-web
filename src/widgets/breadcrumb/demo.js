import * as React from 'react';
import Breadcrumb from './index';
import styled from 'styled-components';
import Icon from '../icon';
import { HashRouter as Router, Route, Switch, Link, withRouter } from 'react-router-dom';

const Hr = styled.div`
  height: 2px;
  background: orange;
  margin: 20px 0;
`;

const CommonIcon = styled(Icon)`
  color: #000;
  border-radius: 50%;
`;

const routes = [
  {
    path: 'index',
    breadcrumbName: '首页',
  },
  {
    path: 'first',
    breadcrumbName: '一级面包屑',
  },
  {
    path: 'second',
    breadcrumbName: '二级面包屑',
  },
  {
    path: 'third',
    breadcrumbName: '当前页面',
  },
];

export default class Demo extends React.Component<any, any> {
  render() {
    return (
      <div>
        {/*Breadcrumb 和 Breadcrumb.Item配合使用，使用href属性时，可跳转 */}
        <Breadcrumb separator={'>'}>
          <Breadcrumb.Item href="">主页</Breadcrumb.Item>
          <Breadcrumb.Item href="">一级菜单</Breadcrumb.Item>
          <Breadcrumb.Item href="">二级菜单</Breadcrumb.Item>
          <Breadcrumb.Item>三级菜单</Breadcrumb.Item>
        </Breadcrumb>
        <Hr />
        {/* 传入routes路由属性，使用默认的 renderItem函数,生成面包屑组件 */}
        <Breadcrumb separator={'>'} routes={routes} />
        <Hr />
        {/* 传入Icon图标 */}
        <Breadcrumb>
          <Breadcrumb.Item href="">
            <CommonIcon iconClass="lugia-icon-logo_chrome" type="home" />
            <span>一级面包屑菜单 </span>
          </Breadcrumb.Item>
          <Breadcrumb.Item href="">
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-financial_smile" type="user" />
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <span>二级面包屑菜单</span>
            <CommonIcon iconClass="lugia-icon-logo_apple" type="user" />
          </Breadcrumb.Item>
        </Breadcrumb>
      </div>
    );
  }
}

// //       demo4
// // !!!!!!!!!!!!!!!!!!!这个组件这么使用有bug，不懂什么原因
// // 使用的自定义设置的 renderItem回调函数,传入routes路由属性，生成面包屑组件=====================
// import * as React from 'react';
// import Breadcrumb from './index';
// import { Link, Router } from 'react-router-dom';

// const routes = [
//   {
//     path: 'index',
//     breadcrumbName: '首页',
//   },
//   {
//     path: 'first',
//     breadcrumbName: '一级面包屑',
//   },
//   {
//     path: 'second',
//     breadcrumbName: '二级面包屑',
//   },
//   {
//     path: 'third',
//     breadcrumbName: '当前页面',
//   },
// ];

// function renderItem(route, params, routes, paths) {
//   const last = routes.indexOf(route) === routes.length - 1;
//   return last ? (
//     <span>{route.breadcrumbName}</span>
//   ) : (
//     <Link to={paths.join('/')}>{route.breadcrumbName}</Link>
//   );
// }
// export default class Demo extends React.Component<any, any> {
//   render() {
//     return (
//       <Router>
//         <Breadcrumb renderItem={renderItem} routes={routes} />
//       </Router>
//     );
//   }
// }

//     demo6
//  给路由传入params 参数
// import * as React from 'react';
// import Breadcrumb from './index';
// const routes = [
//   {
//     name: 'home',
//     breadcrumbName: 'Home',
//     path: '/',
//   },
//   {
//     name: 'apps',
//     breadcrumbName: 'Application List',
//     path: 'apps',
//   },
//   {
//     name: 'app',
//     breadcrumbName: 'Application:id',
//     path: ':id',
//   },
//   {
//     name: 'detail',
//     breadcrumbName: 'Detail',
//     path: 'detail',
//   },
// ];

// export default class Demo extends React.Component<any, any> {
//   render() {
//     return <Breadcrumb params={{ id: 1, class: 2 }} routes={routes} />;
//   }
// }
