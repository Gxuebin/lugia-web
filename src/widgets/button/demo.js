/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import Button from './index';

export default () => {
  return <Button type="danger">hello</Button>;
};
