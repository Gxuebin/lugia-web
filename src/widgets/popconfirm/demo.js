/**
 *
 * create by liangguodong on 2018/12/5
 *
 * @flow
 */
import * as React from 'react';
import Direction from '../button';
import styled from 'styled-components';
import Popconfirm from './popconfirm';

const Wrapper = styled.div`
  margin: 100px;
`;

export default () => {
  const text = 'this is title ';
  const description = 'this is description';
  return (
    <Wrapper>
      <div style={{ marginLeft: 70, whiteSpace: 'nowrap' }}>
        <Popconfirm arrowPosition="topLeft" title={text} action={'click'}>
          <Direction>TL</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="top"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Top</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="topRight"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>TR</Direction>
        </Popconfirm>
      </div>
      <div style={{ width: 70, float: 'left' }}>
        <Popconfirm
          arrowPosition="leftTop"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>LT</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="left"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Left</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="leftBottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>LB</Direction>
        </Popconfirm>
      </div>
      <div style={{ width: 70, marginLeft: 200 }}>
        <Popconfirm
          arrowPosition="rightTop"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>RT</Direction>
        </Popconfirm>
        <Popconfirm arrowPosition="right" title={text}>
          <Direction>Right</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="rightBottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>RB</Direction>
        </Popconfirm>
      </div>
      <div style={{ marginLeft: 70, clear: 'both', whiteSpace: 'nowrap' }}>
        <Popconfirm
          arrowPosition="bottomLeft"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>BL</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="bottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Bottom</Direction>
        </Popconfirm>
        <Popconfirm
          arrowPosition="bottomRight"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>BR</Direction>
        </Popconfirm>
      </div>
      <br />
      <Popconfirm
        arrowPosition="bottom"
        title={text}
        action={'focus'}
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction>聚焦</Direction>
      </Popconfirm>
      <Popconfirm
        arrowPosition="bottom"
        title={text}
        action={'hover'}
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction> 悬停</Direction>
      </Popconfirm>
      <Popconfirm
        arrowPosition="bottom"
        title={text}
        action={'click'}
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction>点击</Direction>
      </Popconfirm>
      <br />
    </Wrapper>
  );
};
