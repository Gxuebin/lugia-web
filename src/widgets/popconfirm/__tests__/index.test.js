/**
 *
 * create by liangguodong
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { WrapperDemo } from '../demo';
import 'jest-styled-components';

describe('popconfirm Demo', () => {
  it('Component JSON', () => {
    global.svtest = true;
    const renders = renderer.create(<WrapperDemo />);
    expect(renders.toJSON()).toMatchSnapshot();
    global.svtest = false;
  });
});
