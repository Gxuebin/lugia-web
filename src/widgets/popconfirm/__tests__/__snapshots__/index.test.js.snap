// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popconfirm Demo Component JSON 1`] = `
.c1 {
  display: inline-block;
  margin-bottom: 0;
  box-sizing: border-box;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  line-height: 1;
  font-family: Trebuchet Ms,Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  position: relative;
  text-transform: none;
  outline: 0;
  color: #666666;
  background-color: #ffffff;
  border: 1px solid #cccccc;
  padding: 0 1.2857142857142858em;
  font-size: 1.4rem;
  border-radius: 0.28571428571428575em;
}

.c1:hover {
  color: #8090ff;
  border: 1px solid #8090ff;
  background-color: rgba(77,99,255,0.05);
}

.c1:focus {
  color: #8090ff;
  border: 1px solid #8090ff;
  background-color: rgba(77,99,255,0.05);
}

.c1:active {
  color: #3e4fcc;
  border: 1px solid #3e4fcc;
}

.c2 {
  display: inline-block;
  line-height: 2.142857142857143em;
}

.c4 {
  line-height: 1.5;
  display: inline-block;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  width: 20rem;
  height: 3.2rem;
  cursor: text;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c4::-webkit-input-placeholder {
  color: #ccc;
}

.c4::-moz-placeholder {
  color: #ccc;
}

.c4:-ms-input-placeholder {
  color: #ccc;
}

.c4::placeholder {
  color: #ccc;
}

.c3 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 {
  margin: 100px;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "marginLeft": 50,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        TL
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        Top
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        TR
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "float": "left",
        "width": 70,
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        LT
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        Left
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        LB
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "marginLeft": 200,
        "width": 70,
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        RT
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        Right
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        RB
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "marginLeft": 50,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        BL
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        Bottom
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        BR
      </span>
    </button>
  </div>
  <br />
  <div
    className="c3"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  >
    <input
      autoFocus={false}
      className="c4"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      placeholder="聚焦弹出"
      size="default"
      style={
        Object {
          "color": "#333333",
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1rem",
          "paddingRight": "3.5rem",
          "paddingTop": "0rem",
        }
      }
      value=""
    />
  </div>
  <button
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
  >
    <span
      className="c2"
      size="default"
    >
       悬停
    </span>
  </button>
  <button
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
  >
    <span
      className="c2"
      size="default"
    >
      点击
    </span>
  </button>
  <br />
  <div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        提示
      </span>
    </button>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
    >
      <span
        className="c2"
        size="default"
      >
        危险操作
      </span>
    </button>
  </div>
</div>
`;
